catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -155,68 +171,102 @@ private boolean isNitfMimeType(String rawMimeType) {
-    contentItems.addAll(newContentItems);
+  public void setMaxSideLength(int maxSideLength) {
+    if (maxSideLength > 0) {
+      LOGGER.trace("Setting derived image maxSideLength to {}", maxSideLength);
+      this.maxSideLength = maxSideLength;
+    } else {
+      LOGGER.debug(
+              "Invalid `maxSideLength` value [{}], must be greater than zero. Default value [{}] will be used instead.",
+              maxSideLength,
+              DEFAULT_MAX_SIDE_LENGTH);
+      this.maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
+    }
+  }
+
+  public void setMaxNitfSizeMB(int maxNitfSizeMB) {
+    this.maxNitfSizeMB = maxNitfSizeMB;
+  }
+
+  public void setCreateOverview(boolean createOverview) {
+    this.createOverview = createOverview;
+  }
+
+  public void setStoreOriginalImage(boolean storeOriginalImage) {
+    this.storeOriginalImage = storeOriginalImage;
   }
 
-  private void process(ContentItem contentItem, List<ContentItem> contentItems) {
-    Metacard metacard = contentItem.getMetacard();
+  NitfRenderer getNitfRenderer() {
+    return new NitfRenderer();
+  }
 
-    if (!isNitfMimeType(contentItem.getMimeTypeRawData())) {
+  private void process(List<ProcessCreateItem> processCreateItems) {
+    List<ProcessCreateItem> newProcessCreateItems = new ArrayList<>();
+    processCreateItems.forEach(processCreateItem -> process(processCreateItem, newProcessCreateItems));
+    processCreateItems.addAll(newProcessCreateItems);
+  }
+
+  private void process(ProcessCreateItem processCreateItem, List<ProcessCreateItem> processCreateItems) {
+    Metacard metacard = processCreateItem.getMetacard();
+    ProcessResource processResource = processCreateItem.getProcessResource();
+
+    String mimeType = processResource.getMimeType();
+
+    if (!isNitfMimeType(mimeType)) {
       LOGGER.debug(
-          "skipping content item: filename={} mimeType={}",
-          contentItem.getFilename(),
-          contentItem.getMimeTypeRawData());
+          "skipping content item: name={} mimeType={}",
+          processResource.getName(),
+          processResource.getMimeType());
       return;
     }
 
-    try {
-      if (contentItem.getSize() / MEGABYTE > maxNitfSizeMB) {
-        LOGGER.debug(
-            "Skipping large ({} MB) content item: filename={}",
-            contentItem.getSize() / MEGABYTE,
-            contentItem.getFilename());
-        return;
-      }
+    if (processResource.getSize() / MEGABYTE > maxNitfSizeMB) {
+      LOGGER.debug(
+              "Skipping large ({} MB) content item: name={}",
+              processResource.getSize() / MEGABYTE,
+              processResource.getName());
+      return;
+    }
+
+    try (FileBackedOutputStream fbos = new FileBackedOutputStream(DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD)){
 
-      BufferedImage renderedImage = renderImage(contentItem);
+      fbos.write(IOUtils.toByteArray(processResource.getInputStream()));
```

Comment:
❓Not directly related to this PR but from what I could tell it looks like `processResource` always returns the same `InputStream`, which means that once it's been consumed nothing else can read from it. This feels a little dangerous, especially since we could potentially have multiple plugins using the resource's `InputStream`. If you confirm that this is actually a potential issue, it may be worth creating a ticket for it.
❓Who should close the `InputStream` once it's been retrieved and consumed?